volatility3/framework/plugins/windows/idt.py

```diff
@@ -0,0 +1,295 @@
+            "Data RO Ac",
+            "Data RW",
+            "Data RW Ac",
+            "Data RO E",
+            "Data RO EA",
+            "Data RW E",
+            "Data RW EA",
+            "Code EO",
+            "Code EO Ac",
+            "Code RE",
+            "Code RE Ac",
+            "Code EO C",
+            "Code EO CA",
+            "Code RE C",
+            "Code RE CA",
+            "<Reserved>",
+            "TSS16 Avl",
+            "LDT",
+            "TSS16 Busy",
+            "CallGate16",
+            "TaskGate",
+            "Int Gate16",
+            "TrapGate16",
+            "<Reserved>",
+            "TSS32 Avl",
+            "<Reserved>",
+            "TSS32 Busy",
+            "CallGate32",
+            "<Reserved>",
+            "Int Gate32",
+            "TrapGate32",
+        ]
+    )
+)
+
+
+class _KIDT:
+    def __init__(self, idt_struct):
+        self.idt = idt_struct
+        self.Offset = idt_struct.Offset
+        self.Selector = idt_struct.Selector
+        self.Access = idt_struct.Access
+        self.ExtendedOffset = idt_struct.ExtendedOffset
+
+    @property
+    def Address(self):
+        if self.ExtendedOffset:
+            return self.ExtendedOffset << 16 | self.Offset
+
+        return 0
+
+
+class _KGDT:
+    def __init__(self, gdt_struct):
+        self.gdt = gdt_struct
+        self.LimitLow = gdt_struct.LimitLow
+        self.BaseLow = gdt_struct.BaseLow
+        self.HighWord = gdt_struct.HighWord
+
+    @property
+    def Type(self):
+        """Get a string name of the descriptor type"""
+        flag = self.HighWord.Bits.Type & 1 << 4
+        typeval = self.HighWord.Bits.Type & ~(1 << 4)
+
+        if flag == 0:
+            typeval += 16
+
+        return GDT_DESCRIPTORS.get(typeval, "UNKNOWN")
+
+    @property
+    def Base(self):
+        """Get the base (start) of memory for this GDT"""
+        return self.BaseLow + (
+            (self.HighWord.Bits.BaseMid + (self.HighWord.Bits.BaseHi << 8)) << 16
+        )
+
+
+class _KPCR:
+    def __init__(self, kpcr_obj, ntkrnlmp, layer_name, symbol_table):
+        self.kpcr = kpcr_obj
+        self.ntkrnlmp = ntkrnlmp
+        self.layer_name = layer_name
+        self.symbol_table = symbol_table
+
+    def idt_entries(self):
+        base_idt = self.kpcr.IDT
+        idt_index = 0
+        for idt_index in range(256):
+            idt_offset = base_idt + 8 * idt_index
+            idt_struct = self.ntkrnlmp.object(
+                object_type="_KIDTENTRY",
+                layer_name=self.layer_name,
+                offset=idt_offset,
+                absolute=True,
+            )
+            try:
+                yield idt_index, _KIDT(idt_struct)
+            except:
+                pass
+
+    def gdt_entries(self):
+        base_gdt = self.kpcr.GDT
+
+        # Since the real GDT size is read from a register, we'll just assume
+        # that there are 128 entries (which is normal for most OS)
+        for gdt_index in range(128):
```

Comment:
Same comment as above regarding the use of `array`. The code here isn't wrong, it could just be simplified a bit and remove some hard coded values.